### PR TITLE
Represent Uri as string, not destructure it, closes #46

### LINKED ISSUE #46: Do not destructure Uri objects
When logging exception data, write Uri object as string, not destructure it.

### PR DESCRIPTION
**Pull request by shatl**

### CODE REVIEW

File: Source/Serilog.Exceptions/Destructurers/ReflectionBasedDestructurer.cs [hunk 1]
```diff
@@ -110,6 +115,22 @@ private object DestructureValueEnumerable(object value, int level, IDictionary<o
             return resultList;
         }
 
+        private object DestructureUri(Uri value, IDictionary<object, IDictionary<string, object>> destructuredObjects, ref int nextCyclicRefId)
+        {
+            if (destructuredObjects.ContainsKey(value))
```
Comment (krajek) [inline Source/Serilog.Exceptions/Destructurers/ReflectionBasedDestructurer.cs]: There are no tests for cyclic reference case.

File: Tests/Serilog.Exceptions.Test/Destructurers/UriDestructurerTest.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
+namespace Serilog.Exceptions.Test.Destructurers
+{
+    using System;
+    using System.Collections;
+    using Core;
+    using Exceptions.Destructurers;
+    using Xunit;
+
+    public class UriDestructurerTest
```
Comment (krajek) [inline Tests/Serilog.Exceptions.Test/Destructurers/UriDestructurerTest.cs]: Personally, I would place the tests in one of the existing suites, preferable in the suite for reflection based destructurer. No other type of object has its own suite, Uri will stick out.

Comment (RehanSaeed) [reply Tests/Serilog.Exceptions.Test/Destructurers/UriDestructurerTest.cs]: Agreed.

**Comment by RehanSaeed** [comment]
@krajek I've added you as a contributor to this project since you've contributed so much and your ideas seem sensible. With great power comes great responsibility!

For some reason this project has suddenly recieved lots of new issues and PR's after a period of inactivity. As the sole founder and maintainer of this project, I've found it hard to keep up and don't want progress to slow down just because I'm working on other things and there is a delay in my response. I will however do my best to continue to review and comment on issues and PR's.

**Comment by krajek** [comment]
Thanks, @RehanSaeed. I really like this project, it helped my commercial projects a lot. After @shatl will implement requested changes, I will merge this PR as my first.

**Comment by krajek** [comment]
Thanks. Actually, the code handling the cycles was correct. There is a crazy scenario when `Uri` is subclassed and then because it can have any "children" it can be a part of a cycle. I will prepare a fix this unlikely scenario later.

**Pull request merged**